src/Magnum/Platform/EmscriptenApplication.h

```diff
@@ -0,0 +1,1509 @@
+        class InputEvent;
+        class MouseEvent;
+        class MouseMoveEvent;
+        class MouseScrollEvent;
+        class KeyEvent;
+        class TextInputEvent;
+        class TextEditingEvent;
+
+        /**
+         * @brief Execute main loop
+         * @return Value for returning from @cpp main() @ce
+         *
+         * Calls @ref mainLoopIteration() in a loop until @ref exit() is
+         * called. See @ref MAGNUM_EMSCRIPTENAPPLICATION_MAIN() for usage
+         * information.
+         */
+        int exec();
+
+        /**
+         * @brief Exit application main loop
+         * @param exitCode  The exit code the application should return
+         *
+         * Stops main loop started by @ref exec().
+         */
+        void exit(int exitCode = 0);
+
+        /**
+         * @brief Run one iteration of application main loop
+         *
+         * Called internally from @ref exec(). If you want to have better
+         * control over how the main loop behaves, you can call this function
+         * yourself from your own `main()` function instead of it being called
+         * automatically from @ref exec() / @ref MAGNUM_EMSCRIPTENAPPLICATION_MAIN().
+         */
+        void mainLoopIteration();
+
+        #ifdef MAGNUM_TARGET_GL
+        /**
+         * @brief Construct with given configuration for OpenGL context
+         * @param arguments         Application arguments
+         * @param configuration     Application configuration
+         * @param glConfiguration   OpenGL context configuration
+         *
+         * Creates application with default or user-specified configuration.
+         * See @ref Configuration for more information. The program exits if
+         * the context cannot be created, see @ref tryCreate() for an
+         * alternative.
+         *
+         * @note This function is available only if Magnum is compiled with
+         *      @ref MAGNUM_TARGET_GL enabled (done by default). See
+         *      @ref building-features for more information.
+         */
+        explicit EmscriptenApplication(const Arguments& arguments, const Configuration& configuration, const GLConfiguration& glConfiguration);
+        #endif
+
+        /**
+         * @brief Construct with given configuration
+         *
+         * Equivalent to calling @ref EmscriptenApplication(const Arguments&, const Configuration&, const GLConfiguration&)
+         * with default-constructed @ref GLConfiguration.
+         */
+        explicit EmscriptenApplication(const Arguments& arguments, const Configuration& configuration);
+
+        /**
+         * @brief Construct with default configuration
+         *
+         * Equivalent to calling @ref EmscriptenApplication(const Arguments&, const Configuration&)
+         * with default-constructed @ref Configuration.
+         */
+        explicit EmscriptenApplication(const Arguments& arguments);
+
+        /**
+         * @brief Construct without creating a window
```

Comment:
Can you replace all occurences of "window" with "setting up a canvas"? It's misleading. And while at it, WebGL instead of OpenGL.

Reply:
Okay, I tried to clarify the docs in this way and added a note to `windowSize` to explain that it is named `windowSize` because of API compat with other Platforms, but retrieves the canvas size.